index.js

```diff
@@ -33,30 +56,12 @@ try {
   await page.keyboard.type(selectors.password);
   await page.click(selectors.login_button);
   await page.waitFor(2000);
+  if (page.url() == selectors.sec_key_tfa_url) {
+    await page.goto(selectors.otp_tfa_url);
```

Comment:
I deleted this since the only thing that seems different when an otp is required is the URL.
Please can you make these changes so that I can merge them. Or if you want me to I can merge them as well.
I really appreciate your time and effort in identifying this issue and coming up with a solution.

Reply:
Wait please, 

If a GitHub user did not set two factor authentication on their account and they login through an unrecognized browser (just as the headless Chrome), GitHub demands for an OTP. The OTP URL is https://github.com/sessions/verified-device which you correctly exported from selectors as `otp_url`

But that URL is not the OTP URL for a GitHub user that has set two factor authentication on their account. If the user set TFA to just send the code to their phone number, then the OTP is demanded at https://github.com/sessions/two-factor and not /sessions/verified-device as in the case of a user without TFA.

Reply:
You're right, my code is wrong. Because it checks for only TFA with security device. Let me explain. 

In addition to TFA with OTP,  GitHub supports a security key option. The security key is a bluetooth or USB external device through which authentication is done. Google accepts that for TFA too. Your mobile phone's fingerprint can also serve as this security key. If a user has set security key in addition to OTP TFA in their GitHub account settings,  GitHub first shows them the login with security key option. 

You can try this. Set security key on your GitHub account when signed in on your mobile phone. Change the headless option of chrome to false (so that it opens the browser), call `node index <your username> <password> <repo-name>`, Chromium will open and input your username and password and login successfully as expected.

 But the next challenge is not OTP. Infact, the URL will be https://github.com/sessions/two-factor/security-key On the page with this URL, there's a link to enter OTP instead. That link takes the user back to the OTP TFA URL, that is, https://github.com/sessions/two-factor (without /security-key)

Reply:
With this in mind, I just called browser to navigate to the second OTP TFA URL with this.
```suggestion
    await page.goto(selectors.otp_tfa_url);
```
And given that the OTP verification logic you @sammychinedu2ky had implemented for plain OTP is the same as the logic for OTP TFA, I moved it to another function and then called it both when there's a normal OTP challenge or there's an OTP TFA challenge

Reply:
Besides, I feel like if I had navigated to `selectors.otp_url`, that is https://github.com/sessions/verified-device , there could be a problem, since the user has set TFA and we are not demanding it from /sessions/two-factor. Kind of who knows 🤷, GitHub could detect an unusual user flow. But this is just my thinking.

Reply:
I had mentioned I was wrong... This is what I mean. I should also have checked if the browser's URL was TFA OTP (the code I wrote only checked for TFA SEC KEY), and then call the verification logic